File: tests/bench/mem_copy.c
/* mem_copy.c — 与 tests/bench/mem_copy.su 对齐（while 两遍扫描 × rounds，无 I/O） */
#include <stdint.h>

int main(void) {
  enum { n = 4096, rounds = 8192 };
  uint8_t buf[n];
  int32_t r;
  int32_t i;
  int32_t j;
  int32_t sum = 0;
  r = 0;
  while (r < rounds) {
    i = 0;
    while (i < n) {
      buf[i] = (uint8_t)i;
      i = i + 1;
    }
    j = 0;
    while (j < n) {
      sum += (int32_t)buf[j];
      j = j + 1;
    }
    r = r + 1;
  }
  return (int)sum;
}

File: tests/bench/mem_copy.zig
// mem_copy.zig — 与 tests/bench/mem_copy.su / mem_copy.c 等价的 Zig 参照
const std = @import("std");

pub fn main() !void {
    const n: usize = 4096;
    var buf: [4096]u8 = undefined;
    var i: usize = 0;
    while (i < n) : (i += 1) {
        buf[i] = @truncate(i);
    }
    var j: usize = 0;
    var sum: i32 = 0;
    while (j < n) : (j += 1) {
        sum += @intCast(buf[j]);
    }
    std.process.exit(@intCast(sum));
}
